### PR TITLE
patch : metric mapping for DI and Performance metrics, to support backwards compatibility from v23.7+

### PR DESCRIPTION
**Pull request by bhalder**
### What does this PR do?
The naming conventions of DI violations and Performance metrics were changed in 23.7
This PR ensures that compatibility with v1 metric names is maintained by adding a compatibility function.

### Motivation
Continue supporting users who have setup their DD charts using the old naming convention.

FDL-17455

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [x] If this PR includes a log pipeline, please add a description describing the remappers and processors. 

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by bhalder** [comment]
@nmuesch can you please help with the reviews here? :)

**Comment by bhalder** [comment]
@iliakur tagging you since you helped review the last PR on this 
Can you please help with reviewing this 🙏

**Comment by emarsha94** [comment]
Hi @bhalder - Thanks for opening this PR! Apologies for the delay. It looks like some of the metrics are changing. If this is the case, can you please make sure you update the dashboard if the dashboard uses the metrics that are being changed?

**Comment by bhalder** [comment]
> Hi @bhalder - Thanks for opening this PR! Apologies for the delay. It looks like some of the metrics are changing. If this is the case, can you please make sure you update the dashboard if the dashboard uses the metrics that are being changed?

Thanks for taking time to take a look.
The dashboards do not use the metrics being updated.

**Review by emarsha94 — APPROVED** [review]

File: fiddler/datadog_checks/fiddler/check.py [hunk 1]
```diff
@@ -169,10 +169,27 @@ def _process_v1compat_query_result(
         value_list = result['data'][-1] if result['data'] else 0
         gauge_results = []
 
-        # Update metric names if necessary
-        metric_mapping = {'traffic': 'traffic_count', 'jsd': 'histogram_drift'}
+        # metric name mapping from 23.7 -> 23.5 format
+        metric_mapping = {
+            'traffic': 'traffic_count',
+            'jsd': 'histogram_drift',
+            'expected_calibration_error': 'expected_callibration_error',
```
Comment (steveny91) [inline fiddler/datadog_checks/fiddler/check.py:176]: Are we purposely renaming this metric to a typo? Or is `callibration` a specific term for used fiddler?

Comment (bhalder) [reply fiddler/datadog_checks/fiddler/check.py:176]: This is a typo in the older version of Fiddler which has been fixed.
This map ensures backwards compatibility with older Dashboards that customers are on.

Comment (steveny91) [reply fiddler/datadog_checks/fiddler/check.py:176]: I guess would it be better to update to fix the typo in the new version and then sync the dashboard to use the corrected version? @emarsha94 what do you think? I'm okay with merging this as is.

Comment (emarsha94) [reply fiddler/datadog_checks/fiddler/check.py:176]: To confirm, the metric has a typo so we updated the metric to fix the type. However, the old metric is still being used? If it's not used moving forward then we should update the dashboard with the new version. Customers can pull previous metric history on the old metric as needed. You can include two metrics in a graph widget if you want customers to see both.

Comment (bhalder) [reply fiddler/datadog_checks/fiddler/check.py:176]: Rechecked dashboard usage. There is no consumer of this metric.
The default dashboard available on Datadog does not use this metric either.

I think, we should be fine.

**Review by nmuesch — APPROVED** [review]
Looks good from the manifest + scaffolding side.

File: fiddler/datadog_checks/fiddler/check.py (hunk 1 above)
Comment (steveny91) [reply fiddler/datadog_checks/fiddler/check.py:176]: @emarsha94 Yeah basically that. Older version of the fiddler has a typo in the metric name and the new version of fiddler fixed it. The line I'm commenting on remaps the corrected metric name to the older typo version.

I'll approve this PR as this is more of a design choice than a coding issue at this point. I'll go with whatever you both decide on.

**Review by steveny91 — APPROVED** [review]

File: fiddler/datadog_checks/fiddler/check.py (hunk 1 above)
Comment (bhalder) [reply fiddler/datadog_checks/fiddler/check.py:176]: Nope. The default dashboard does not use it.

**Pull request merged**